### ISSUE TITLE
Restoring parent server’s files deletes sub servers’ files (this bug causes data loss)

### ISSUE THREAD

**Issue by dtwist** [virtualmin/virtualmin-gpl#294]
[ I've posted this as a [question on the forum](https://forum.virtualmin.com/t/how-to-restore-a-parent-servers-files-and-db-without-deleting-sub-servers/109853), but have not gotten any replies, so filing as a bug, since it causes unexpected data loss. ]

@jcameron @swelljoe : Pinging you both, since the Git revision history has your names all over the backup/restore functionality.

### Brief:
When restoring a parent server's files, regardless of the restore options chosen, that parent's child servers' files are always deleted.

### Steps to Reproduce:
[note] I haven't tested to see if it matters, but I am backing up to and restoring from AWS S3.

1. Create a backup of a parent server (that has children) using "one file per server" option and backup all features. Do not check the option to include sub-servers in parent servers’ backups.

2. Restore the parent server from that backup. Do not tick the box “Delete files that are not in the backup”. Choose to restore only files, and do NOT select the option to delete and rebuild the virtual server.

### Result:
The sub-servers’ files are all deleted.

### Expected Result:
The sub-servers’ files should remain.

Operating system: CentOS
OS version: 7.9.2009

**Comment by jcameron** [comment]
I just tested this, and wasn't able to re-create the problem.

Where under the sub-server's directory are the files that were deleted?

**Comment by dtwist** [comment]
I’m not at my computer right now, but IIRC, everything beneath the children’s folders (e.g. /home/parent-server/domains/child-domain/*) is wiped out.

**Comment by dtwist** [comment]
Back at my computer, and I tried a test using local storage, but all the same settings as setup for scheduled backups to AWS S3. As in your test, the bug did not occur for me. So, it seems the issue is specific to AWS S3 backup/restore. I'll let a backup cycle happen tonight, then try restoring the test parent/child servers I created via AWS and let you know what happens.

**Comment by jcameron** [comment]
Thanks, please do.

**Comment by dtwist** [comment]
Hi Jamie,

I waited for a full backup to complete on schedule, so I could be sure all conditions are the same for the servers we've had issues with. This test came back without issues too though, so I tried with an existing parent server that had exhibited the bug previously. That worked this time too… 

I don't know what has changed, but the issue seems to have cleared up since I reported it. (I am certain it was an issue then though, as we tested thoroughly and with several parent servers.) Let's mark this bug closed, and I'll report back if we encounter it again in future.

Thanks,
David
